devops/ansible/roles/girder/library/girder.py

```diff
@@ -942,10 +950,15 @@ def update(self, _id, body, **kwargs):
         if _id in self.resources:
             current = self.resources[_id]
             # if body is a subset of current we don't actually need to update
-            if set(body.items()) <= set(current.items()):
-                return current
-            else:
-                return self.__apply(_id, self.client.put, body, **kwargs)
+            try:
+                if set(body.items()) <= set(
```

Comment:
It is the dictionary of values we are setting on the resource.  I don't know why it is called `body`, and I haven't checked that we can't pass non-string values in the ansible yml.  It wouldn't surprise me if we can.

The `try`/`except` will throw in this case and we may needlessly update the resource when it matches the desired state, but only wastes a Mongo call rather than breaks.